proj_eclipse/src/main/java/smartmag/models/ui/ProductTableModel.java

```diff
@@ -16,15 +16,22 @@ public class ProductTableModel extends AbstractTableModel
 	private String[] columnNames;
 	private TreeMap<Integer, ProductModel> prodotti;
 
+	/**
+	 * costruttore
+	 */
 	public ProductTableModel() {
 		refreshFromModel();
 		this.columnNames = new String[] { "Id", "Nome", "Descrizione", "Peso",
 				"Soglia" };
 		ProductModel.addChangeListener(this);
 	}
 
-	private void refreshFromModel() {
+	/**
+	 * aggiorna la lista di prodotti
+	 */
+	public void refreshFromModel() {
 		prodotti = ProductModel.getAllProductModels();
+		fireTableDataChanged();
```

Comment:
C'è già `stateChanged()` per forzare aggiornamento dall'esterno e richiama già lui il `notifyChangeListeners`.
```suggestion
	/**
	 * aggiorna la lista di prodotti
	 */
	private void refreshFromModel() {
		prodotti = ProductModel.getAllProductModels();
```